python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -707,18 +708,56 @@ def _submit_runs_and_update_backfill_in_chunks(
             break
 
         # Submit runs in the chunk
-        for run_request in run_requests_chunk:
+        run_requests_i = 0
+        while run_requests_i < len(run_requests_chunk):
+            run_request = run_requests_chunk[run_requests_i]
             yield None
-            submit_run_request(
-                run_request=run_request,
-                asset_graph=asset_graph,
-                # create a new request context for each run in case the code location server
-                # is swapped out in the middle of the backfill
-                workspace=workspace_process_context.create_request_context(),
-                instance=instance,
-                pipeline_and_execution_plan_cache=pipeline_and_execution_plan_cache,
+
+            # create a new request context for each run in case the code location server
+            # is swapped out in the middle of the backfill
+            workspace = workspace_process_context.create_request_context()
+            execution_data = get_job_execution_data_from_run_request(
+                asset_graph,
+                run_request,
+                instance,
+                workspace=workspace,
+                run_request_execution_data_cache=run_request_execution_data_cache,
             )
 
+            if _execution_plan_targets_asset_selection(
+                execution_data.external_execution_plan.execution_plan_snapshot,
+                check.not_none(run_request.asset_selection),
+            ):
+                submit_run_request(
+                    run_request=run_request,
+                    workspace=workspace,
+                    instance=instance,
+                    run_request_execution_data=execution_data,
+                )
+                run_requests_i += 1
+
+            elif num_retries_allowed > 0:
+                logger.warning(
+                    "Execution plan is out of sync with the workspace. Pausing the backfill for "
+                    "60s to allow the execution plan to rebuild with the updated workspace."
```

Comment:
nit: make this `{RELOAD_WORKSPACE_INTERVAL}s` instead of 60s

Reply:
ahh this is a good call